lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -303,33 +305,56 @@ toSlotNo :: W.EpochLength -> W.SlotId -> SlotNo
 toSlotNo epLength =
     SlotNo . W.flatSlot epLength
 
+toBlockHeader
+    :: W.Hash "Genesis"
+    -> W.EpochLength
+    -> ShelleyBlock
+    -> W.BlockHeader
+toBlockHeader genesisHash epLength blk =
+    let
+        O.ShelleyBlock (SL.Block (SL.BHeader header _) _) headerHash = blk
+    in
+    W.BlockHeader
+        { slotId =
+            fromSlotNo epLength $ SL.bheaderSlotNo header
+        , blockHeight =
+            fromBlockNo $ SL.bheaderBlockNo header
+        , headerHash =
+            fromShelleyHash headerHash
+        , parentHeaderHash =
+            fromPrevHash (coerce genesisHash) $
+                SL.bheaderPrev header
+        }
+
 fromShelleyBlock
     :: W.Hash "Genesis"
     -> W.EpochLength
     -> ShelleyBlock
     -> W.Block
 fromShelleyBlock genesisHash epLength blk =
     let
-       O.ShelleyBlock (SL.Block (SL.BHeader header _) txSeq) headerHash = blk
+       O.ShelleyBlock (SL.Block _ txSeq) _ = blk
        SL.TxSeq txs' = txSeq
-       (txs, certs) = unzip $ map fromShelleyTx $ toList txs'
+       (txs, certs, _) = unzip3 $ map fromShelleyTx $ toList txs'
 
     in W.Block
-        { header = W.BlockHeader
-            { slotId =
-                fromSlotNo epLength $ SL.bheaderSlotNo header
-            , blockHeight =
-                fromBlockNo $ SL.bheaderBlockNo header
-            , headerHash =
-                fromShelleyHash headerHash
-            , parentHeaderHash =
-                fromPrevHash (coerce genesisHash) $
-                    SL.bheaderPrev header
-            }
+        { header = toBlockHeader genesisHash epLength blk
         , transactions = txs
         , delegations  = mconcat certs
         }
 
+fromShelleyBlock'
+    :: W.EpochLength
+    -> ShelleyBlock
+    -> (W.SlotId, [W.PoolRegistrationCertificate])
```

Comment:
We should extract retirement certificates here too, eventually?

Btw, to me scattering the logic like this feels a bit wrong. Wouldn't matching on the `PoolCert` constructors in the `Shelley.Pool` module directly be clearer? https://github.com/input-output-hk/cardano-ledger-specs/blob/036ff8223310ecdaf7f0ed8c211c691cb0dcb8f3/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxData.hs#L287-L292

Reply:
Ok, I saw your:
> FIXME We need to acknowledge pool retirement